### ISSUE TITLE
[java] SwitchStmtsShouldHaveDefault false-positive with Java 14 switch non-fallthrough branches

### ISSUE THREAD

**Issue by boris-petrov** [pmd/pmd#2806]
**Affects PMD Version:** 

All, including 6.28.0.

**Rule:** [SwitchStmtsShouldHaveDefault](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#switchstmtsshouldhavedefault)

**Description:**

```java
	public String foo(String foo) {
		switch (foo) {
			case "=" -> foo = "eq";
			case "!=" -> foo = "ne";
			default -> throw new IllegalArgumentException();
		}
		return foo + 'a';
	}
```

This causes a `SwitchStmtsShouldHaveDefault` warning.

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

Gradle.

**Comment by adangel** [comment]
This has been fixed with PMD 7.0.0-rc1.
